tests/smoke_test.go

```diff
@@ -452,3 +453,45 @@ func TestPluginRun(t *testing.T) {
 	_, stderr = e.RunCommandExpectError("pulumi", "plugin", "run", "--kind=resource", "random", "--", "--help")
 	assert.Contains(t, stderr, "flag: help requested")
 }
+
+func TestInstall(t *testing.T) {
+	t.Parallel()
+
+	for _, runtime := range Runtimes {
+		// Reassign runtime before capture since it changes while looping.
+		runtime := runtime
+
+		t.Run(runtime, func(t *testing.T) {
+			t.Parallel()
+
+			e := ptesting.NewEnvironment(t)
+			defer deleteIfNotFailed(e)
+
+			// Make sure the random provider is not installed locally
+			// so that we can test the `install` command works.
+			out, _ := e.RunCommand("pulumi", "plugin", "ls")
+			if strings.Contains(out, "random  resource") {
+				e.RunCommand("pulumi", "plugin", "rm", "resource", "random", "--all", "--yes")
+			}
+
+			// `new` wants to work in an empty directory but our use of local url means we have a
+			// ".pulumi" directory at root.
+			projectDir := filepath.Join(e.RootPath, "project")
+			err := os.Mkdir(projectDir, 0o700)
+			require.NoError(t, err)
+
+			e.CWD = projectDir
+
+			e.RunCommand("pulumi", "login", "--cloud-url", e.LocalURL())
+			// Pass `--generate-only` so dependencies are not installed as part of the `new` command.
+			e.RunCommand("pulumi", "new", "random-"+Languages[runtime], "--yes", "--generate-only")
+
+			// Ensure `install` works and subsequent `up` and `destroy` operations work.
+			_, stderr := e.RunCommand("pulumi", "install")
+			assert.Regexp(t, regexp.MustCompile(`\[resource plugin random-.+\] installing`), stderr)
```

Comment:
This test is failing:

https://github.com/pulumi/pulumi/actions/runs/10512638165/job/29126736206#step:36:575

```
=== Failed
=== FAIL: . TestInstall/python (59.20s)
    environment.go:103: Created new test environment:  /tmp/test-env3425907744
    environment.go:223: Running command pulumi plugin ls
    environment.go:223: Running command pulumi login --cloud-url file:///tmp/test-env3425907744
    environment.go:223: Running command pulumi new random-python --yes --generate-only
    environment.go:223: Running command pulumi install
    smoke_test.go:491: 
        	Error Trace:	/home/runner/work/pulumi/pulumi/tests/smoke_test.go:491
        	Error:      	Expect "resource plugin random-4.16.3 version 4.16.3 installing
        	            	" to match "\[resource plugin random-.+\] installing"
        	Test:       	TestInstall/python
    environment.go:223: Running command pulumi stack init test
    environment.go:223: Running command pulumi up --yes
    environment.go:223: Running command pulumi destroy --yes

=== FAIL: . TestInstall/go (55.05s)
    environment.go:103: Created new test environment:  /tmp/test-env948558346
    environment.go:223: Running command pulumi plugin ls
    environment.go:223: Running command pulumi login --cloud-url file:///tmp/test-env948558346
    environment.go:223: Running command pulumi new random-go --yes --generate-only
    environment.go:223: Running command pulumi install
    smoke_test.go:491: 
        	Error Trace:	/home/runner/work/pulumi/pulumi/tests/smoke_test.go:491
        	Error:      	Expect "resource plugin random-4.13.0 version 4.13.0 installing
        	            	" to match "\[resource plugin random-.+\] installing"
        	Test:       	TestInstall/go
    environment.go:223: Running command pulumi stack init test
    environment.go:223: Running command pulumi up --yes
    environment.go:223: Running command pulumi destroy --yes

=== FAIL: . TestInstall/dotnet (66.55s)
    environment.go:103: Created new test environment:  /tmp/test-env315653803
    environment.go:223: Running command pulumi plugin ls
    environment.go:223: Running command pulumi login --cloud-url file:///tmp/test-env315653803
    environment.go:223: Running command pulumi new random-csharp --yes --generate-only
    environment.go:223: Running command pulumi install
    smoke_test.go:491: 
        	Error Trace:	/home/runner/work/pulumi/pulumi/tests/smoke_test.go:491
        	Error:      	Expect "resource plugin random-4.16.3 version 4.16.3 installing
        	            	" to match "\[resource plugin random-.+\] installing"
        	Test:       	TestInstall/dotnet
    environment.go:223: Running command pulumi stack init test
    environment.go:223: Running command pulumi up --yes
    environment.go:223: Running command pulumi destroy --yes

=== FAIL: . TestInstall/nodejs (62.78s)
    environment.go:103: Created new test environment:  /tmp/test-env1095760047
    environment.go:223: Running command pulumi plugin ls
    environment.go:223: Running command pulumi login --cloud-url file:///tmp/test-env1095760047
    environment.go:223: Running command pulumi new random-typescript --yes --generate-only
    environment.go:223: Running command pulumi install
    smoke_test.go:491: 
        	Error Trace:	/home/runner/work/pulumi/pulumi/tests/smoke_test.go:491
        	Error:      	Expect "resource plugin random-4.16.3 version 4.16.3 installing
        	            	" to match "\[resource plugin random-.+\] installing"
        	Test:       	TestInstall/nodejs
    environment.go:223: Running command pulumi stack init test
    environment.go:223: Running command pulumi up --yes
    environment.go:223: Running command pulumi destroy --yes

=== FAIL: . TestInstall/yaml (54.44s)
    environment.go:103: Created new test environment:  /tmp/test-env3963462627
    environment.go:223: Running command pulumi plugin ls
    environment.go:223: Running command pulumi login --cloud-url file:///tmp/test-env3963462627
    environment.go:223: Running command pulumi new random-yaml --yes --generate-only
    environment.go:223: Running command pulumi install
    smoke_test.go:491: 
        	Error Trace:	/home/runner/work/pulumi/pulumi/tests/smoke_test.go:491
        	Error:      	Expect "resource plugin random version 4.16.3 installing
        	            	" to match "\[resource plugin random-.+\] installing"
        	Test:       	TestInstall/yaml
    environment.go:223: Running command pulumi stack init test
    environment.go:223: Running command pulumi up --yes
    environment.go:223: Running command pulumi destroy --yes

=== FAIL: . TestInstall/java (78.93s)
    environment.go:103: Created new test environment:  /tmp/test-env1232564630
    environment.go:223: Running command pulumi plugin ls
    environment.go:223: Running command pulumi login --cloud-url file:///tmp/test-env1232564630
    environment.go:223: Running command pulumi new random-java --yes --generate-only
    environment.go:223: Running command pulumi install
    smoke_test.go:491: 
        	Error Trace:	/home/runner/work/pulumi/pulumi/tests/smoke_test.go:491
        	Error:      	Expect "resource plugin random-4.17.0-alpha.1723010172 version 4.17.0-alpha.1723010172 installing
        	            	" to match "\[resource plugin random-.+\] installing"
        	Test:       	TestInstall/java
    environment.go:223: Running command pulumi stack init test
    environment.go:223: Running command pulumi up --yes
    environment.go:223: Running command pulumi destroy --yes

=== FAIL: . TestInstall (0.00s)
```